nmdc_schema/migrators/migrator_from_X_to_PR10.py

```diff
@@ -0,0 +1,113 @@
+            metagenome_annotation_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetagenomeAnnotation")],
+            metagenome_assembly_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetagenomeAssembly")],
+            metagenome_sequencing_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetagenomeSequencing")],
+            metaproteomics_analysis_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetaproteomicsAnalysis")],
+            metatranscriptome_analysis_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:MetatranscriptomeAnalysis")],
+            nom_analysis_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:NomAnalysis")],
+            data_generation_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:DataGeneration")],
+            pooling_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:Pooling")],
+            processed_sample_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:ProcessedSample")],
+            read_based_taxonomy_analysis_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:ReadBasedTaxonomyAnalysis")],
+            read_qc_analysis_set=[lambda document: self.add_type_slot_with_class_uri(document, "nmdc:ReadQcAnalysis")],
+        )
+
+    def add_type_to_inlined_classes(self, document: dict, slot: str, uri: str):
+        r"""
+        Adds a type slot to each inlined instance of an NMDC class in the biosmpale_set, study_set, and extraction_set collections. 
+        """
+        if document.get(slot):
+            # If slot is a list, iterate over each item in the list (e.g. chem_administration and has_credit_associations)
+            if isinstance(document[slot], list):
+                for item in document[slot]:
+                    # will exclude non-inlined class slots that use an id
+                    if not isinstance(item, str):
+                        item["type"] = uri
+                        if item.get("term"):
+                            item["term"]["type"] = "nmdc:OntologyClass"
+                        if item.get("applies_to_person"):
+                            item["applies_to_person"]["type"] = "nmdc:PersonValue"
+            else:
+                # If slot is not multivalued
+                document[slot]["type"] = uri
+                if document[slot].get("term"):
+                    document[slot]["term"]["type"] = "nmdc:OntologyClass"
+                    
+
+    def add_type_slot_with_class_uri(self, document: dict, class_uri: str, inlined_slots=None):
+        r"""
+            Adds a type slot to each collection with the appropriate class uri as the value. E.g. type: nmdc:Biosample. If a type
+            slot exists, it will overwrite to the types listed below. If the optional dictionary argument inlined_slots is given, then
```

Comment:
@aclum I documented that if a type slot exists, it will overwrite to the correct types.

Reply:
I will look at this deeper to make sure it does do that so any typos will be overwritten.